File: leetcode-04/solutions/python3/1002.py
class Solution:
    def commonChars(self, A: List[str]) -> List[str]:
        ans = []
        commonCount = [float('inf')] * 26

        for a in A:
            count = [0] * 26
            for c in a:
                count[ord(c) - ord('a')] += 1
            for i in range(26):
                commonCount[i] = min(commonCount[i], count[i])

        for i in range(26):
            for j in range(commonCount[i]):
                ans.append(chr(i + ord('a')))

        return ans

File: leetcode-04/solutions/cpp/1002.cpp
class Solution {
 public:
  vector<string> commonChars(vector<string>& A) {
    vector<string> ans;
    vector<int> commonCount(26, INT_MAX);

    for (string& a : A) {
      vector<int> count(26);
      for (char c : a) count[c - 'a']++;
      for (int i = 0; i < 26; i++)
        commonCount[i] = min(commonCount[i], count[i]);
    }

    for (int i = 0; i < 26; i++)
      for (int j = 0; j < commonCount[i]; j++)
        ans.push_back(string(1, i + 'a'));

    return ans;
  }
};